mmengine/config/config.py

```diff
@@ -161,40 +355,59 @@ def __init__(self,
 
         Returns:
             Config: Config instance built from config file.
         """
         filename = str(filename) if isinstance(filename, Path) else filename
-        cfg_dict, cfg_text, env_variables = Config._file2dict(
-            filename, use_predefined_variables, use_environment_variables)
-        if import_custom_modules and cfg_dict.get('custom_imports', None):
-            try:
-                import_modules_from_strings(**cfg_dict['custom_imports'])
-            except ImportError as e:
-                err_msg = (
-                    'Failed to import custom modules from '
-                    f"{cfg_dict['custom_imports']}, the current sys.path is: ")
-                for p in sys.path:
-                    err_msg += f'\n    {p}'
-                err_msg += (
-                    '\nYou should set `PYTHONPATH` to make `sys.path` include '
-                    'the directory which contains your custom module')
-                raise ImportError(err_msg) from e
-        return Config(
-            cfg_dict,
-            cfg_text=cfg_text,
-            filename=filename,
-            env_variables=env_variables)
+        if lazy_import is None:
+            lazy_import = Config._is_lazy_import(filename)
+        if not lazy_import:
+            cfg_dict, cfg_text, env_variables = Config._file2dict(
+                filename, use_predefined_variables, use_environment_variables)
+            if import_custom_modules and cfg_dict.get('custom_imports', None):
+                try:
+                    import_modules_from_strings(**cfg_dict['custom_imports'])
+                except ImportError as e:
+                    err_msg = (
+                        'Failed to import custom modules from '
+                        f"{cfg_dict['custom_imports']}, the current sys.path "
+                        'is: ')
+                    for p in sys.path:
+                        err_msg += f'\n    {p}'
+                    err_msg += (
+                        '\nYou should set `PYTHONPATH` to make `sys.path` '
+                        'include the directory which contains your custom '
+                        'module')
+                    raise ImportError(err_msg) from e
+            return Config(
+                cfg_dict,
+                cfg_text=cfg_text,
+                filename=filename,
+                env_variables=env_variables)
+        else:
+            # enable lazy import when parsing the config
+            ConfigDict.lazy = True
+            cfg_dict, imported_names = Config._parse_lazy_import(filename)
+            for key, value in list(cfg_dict.to_dict().items()):
+                if isinstance(value, (types.FunctionType, types.ModuleType)):
+                    cfg_dict.pop(key)
+            # disable lazy import to get the real type.
+            ConfigDict.lazy = False
+            cfg = Config(cfg_dict, filename=filename)
+            object.__setattr__(cfg, '_imported_names', imported_names)
```

Comment:
```suggestion
            super().__setattr__(cfg, '_imported_names', imported_names)
```

Reply:
We can use `super` here since it is a staticmethod